### ISSUE TITLE
Document how to set up a fork for mdx-deck patching / development

### ISSUE THREAD

**Issue by alpinegizmo** [jxnblk/mdx-deck#372]
I'm struggling to figure out how to successfully use a fork of mdx-deck. I realize this is due to my being a node.js newbie, but having spent about a day struggling to figure out whether to use npm or yarn, where in the source tree to do a `yarn link` (I'm guessing in packages/mdx-deck), why moew is missing, what it means that I have `more than one workspace with name "@mdx-deck/webpack-html-plugin"`, etc., what I wish for is a HOWTO that documents the steps needed to use a fork of mdx-deck. 

mdx-deck is awesome enough to get me to learn a new dev environment, and I hope to contribute something useful back once I get a working dev environment setup.

**Comment by alpinegizmo** [comment]
FWIW, my current status is that I believe I've gotten everything cleanly setup, but when I try `yarn start` it is either stuck, or is taking a really long time. Since there isn't much CPU or network activity, I'm assuming it's stuck.

```
$ npm start

> mdx-experiment@1.0.0 start /Users/david/stuff/mdx-experiment
> mdx-deck deck.mdx

[mdx-deck] starting dev server

● [mdx-deck] █████████████████████████ building (20%) 90/91 modules 1 active
 babel-loader › deck.mdx
```

**Comment by alpinegizmo** [comment]
Possibly related to https://github.com/yarnpkg/yarn/issues/6801

**Comment by jxnblk** [comment]
Hi! Sounds like you've gotten part of the way there. I apologize that the [CONTRIBUTING.md](https://github.com/jxnblk/mdx-deck/blob/master/CONTRIBUTING.md) doc isn't fully up-to-date.

After cloning, you should be able to run `yarn` in the root directory. You'll need to install the Yarn CLI if you don't have it already. The repo uses [Yarn workspaces](https://yarnpkg.com/en/docs/workspaces), which should link the local packages together for local development.

- To start the demo in `docs/demo.mdx`, run `yarn start`
- To run the test suite, `yarn test`

From there, hopefully you can poke around the source code to see how things fit together, but feel free to ask any other questions, or let me know if you're still stuck with the dev setup

**Comment by jxnblk** [comment]
And be sure you're on Node version 8, 10, or 12 – sometimes that can be an issue with development setups like this one

**Comment by alpinegizmo** [comment]
Thanks! Yes, I have the demo and tests running. The pointer to yarn workspaces was helpful. But I'm still stuck. 

What I haven't been able to figure out is how to use my local fork of mdx-deck in own my presentation. I've tried npm link and yarn link and yalc and some others things I've found, but keep having problems. Either I end up with build problems (that generally revolve around @mdx-deck/webpack-html-plugin), or I get an runtime error about hooks and multiple copies of react.

What's the recommended workflow here?

**Comment by jxnblk** [comment]
I don’t know what the best approach for what you’re doing is, but one common way of using a forked package is to rename whatever packages you changed with your username scope on npm and publish it; e.g. for myself, it could be renamed `@jxnblk/mdx-deck`

If you’re planning on opening a PR with changes, that can also be useful for testing things out outside of the repo and not being blocked by the PR

**Comment by jxnblk** [comment]
Closing due to inactivity, but feel free to open a new issue if you're still encountering issues
